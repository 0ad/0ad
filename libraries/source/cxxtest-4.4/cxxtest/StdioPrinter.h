/*
-------------------------------------------------------------------------
 CxxTest: A lightweight C++ unit testing library.
 Copyright (c) 2008 Sandia Corporation.
 This software is distributed under the LGPL License v3
 For more information, see the COPYING file in the top CxxTest directory.
 Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
 the U.S. Government retains certain rights in this software.
-------------------------------------------------------------------------
*/

#ifndef __cxxtest__StdioPrinter_h__
#define __cxxtest__StdioPrinter_h__

//
// The StdioPrinter is an StdioFilePrinter which defaults to stdout.
// This should have been called StdOutPrinter or something, but the name
// has been historically used.
//

#include <cxxtest/StdioFilePrinter.h>

namespace CxxTest
{
class StdioPrinter : public StdioFilePrinter
{
public:
    StdioPrinter(FILE *o = stdout, const char *preLine = ":", const char *postLine = "") :
        StdioFilePrinter(o, preLine, postLine) {}
};
}

#endif // __cxxtest__StdioPrinter_h__
