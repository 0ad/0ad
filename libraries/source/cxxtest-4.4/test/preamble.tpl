// -*- C++ -*-

#define CXXTEST_ABORT_TEST_ON_FAIL

// CxxTest definitions and headers
<CxxTest preamble>

// Make sure this worked
#ifndef TS_ASSERT
#   error The preamble does not work!
#endif

#include <cxxtest/StdioPrinter.h>

int main()
{
    CxxTest::StdioPrinter runner;

    TS_FAIL( "This will not be displayed" );
    int result = runner.run() + runner.run();
    TS_FAIL( "This will not be displayed" );
    
    return result;
}


// The CxxTest "world"
<CxxTest world>

