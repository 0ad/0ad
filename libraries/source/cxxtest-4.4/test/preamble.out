Running cxxtest tests (16 tests)
In CreatedTest::test_nothing:
CreatedTest.h:24: Error: Test failed: Nothing to test
..
In EnumTraits::test_Enum_traits:
EnumTraits.h:30: Error: Test failed: Yes
In ExceptionTest::testAssertion:
ExceptionTest.h:18: Error: Expected (throwThis(5)) to throw (const char *) but it threw something else
In FixtureTest::test_strcpy:
FixtureTest.h:28: Error: Expected (_buffer[1] == 'E'), found ('e' != 'E')
In MessageTest::testValues:
MessageTest.h:21: Error: Test failed: My hovercraft
MessageTest.h:21: Error: Assertion failed: value != 0
In SimpleTest::testEquality:
SimpleTest.h:14: Error: Expected (1 == 2), found (1 != 2)
In SimpleTest::testAddition:
SimpleTest.h:23: Error: Expected (2 + 2 == 5), found (4 != 5)
In SimpleTest::TestMultiplication:
SimpleTest.h:28: Error: Expected (4 * 4 == 44), found (16 != 44)
In SimpleTest::testComparison:
SimpleTest.h:34: Error: Expected (-1 < -2), found (-1 >= -2)
In SimpleTest::testTheWorldIsCrazy:
SimpleTest.h:38: Error: Expected (true == false), found (true != false)
In SimpleTest::test_Failure:
SimpleTest.h:42: Error: Test failed: Not implemented
In SimpleTest::test_TS_SKIP_macro:
SimpleTest.h:47: Warning: Test skipped: Simply skip this test
s
In SimpleTest::test_TS_WARN_macro:
SimpleTest.h:52: Warning: Just a friendly warning
SimpleTest.h:53: Warning: Warnings don't abort the test
.
In TestFunky::testPets:
TraitsTest.h:53: Error: Expected (pet1 == pet2), found (Pet("dog") != Pet("cat"))
Failed 12 and Skipped 1 of 16 tests
Success rate: 20%
Running cxxtest tests (16 tests)
In CreatedTest::test_nothing:
CreatedTest.h:24: Error: Test failed: Nothing to test
..
In EnumTraits::test_Enum_traits:
EnumTraits.h:30: Error: Test failed: Yes
In ExceptionTest::testAssertion:
ExceptionTest.h:18: Error: Expected (throwThis(5)) to throw (const char *) but it threw something else
In FixtureTest::test_strcpy:
FixtureTest.h:28: Error: Expected (_buffer[1] == 'E'), found ('e' != 'E')
In MessageTest::testValues:
MessageTest.h:21: Error: Test failed: My hovercraft
MessageTest.h:21: Error: Assertion failed: value != 0
In SimpleTest::testEquality:
SimpleTest.h:14: Error: Expected (1 == 2), found (1 != 2)
In SimpleTest::testAddition:
SimpleTest.h:23: Error: Expected (2 + 2 == 5), found (4 != 5)
In SimpleTest::TestMultiplication:
SimpleTest.h:28: Error: Expected (4 * 4 == 44), found (16 != 44)
In SimpleTest::testComparison:
SimpleTest.h:34: Error: Expected (-1 < -2), found (-1 >= -2)
In SimpleTest::testTheWorldIsCrazy:
SimpleTest.h:38: Error: Expected (true == false), found (true != false)
In SimpleTest::test_Failure:
SimpleTest.h:42: Error: Test failed: Not implemented
In SimpleTest::test_TS_SKIP_macro:
SimpleTest.h:47: Warning: Test skipped: Simply skip this test
s
In SimpleTest::test_TS_WARN_macro:
SimpleTest.h:52: Warning: Just a friendly warning
SimpleTest.h:53: Warning: Warnings don't abort the test
.
In TestFunky::testPets:
TraitsTest.h:53: Error: Expected (pet1 == pet2), found (Pet("dog") != Pet("cat"))
Failed 12 and Skipped 1 of 16 tests
Success rate: 20%
Error level = 24
